BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m

```diff
@@ -241,38 +250,52 @@ - (void)failAuthenticationWithConnection:(NSURLConnection *)connection
     self.connection = nil;
     self.connectionResponse = nil;
     [self setWebViewCanBeUsedDirectly:NO forHost:connection.currentRequest.URL.host];
-
+    
+    self.connectionError = error;
+    self.connectionErrorMessage = message;
+    
+    // We can only handle the connection error if we're in the view hierarchy. It would be difficult to display UIAlertControllers when not sure we're in the view hierarchy. We will try again after viewDidAppear
     if (self.view.window) {
-        
-        self.connectionError = error;
-        
-        UIAlertController *alertController = [UIAlertController alertControllerWithTitle:NSLocalizedString(@"Unable to Log In", @"Alert view title: Title for failed SSO login due to authentication issue")
-                                                                                 message:message
-                                                                          preferredStyle:UIAlertControllerStyleAlert];
-        
-        __weak typeof(self) weakSelf = self;
-        void (^completion)(UIAlertAction *action) = ^void(UIAlertAction *action) {
-            [alertController dismissViewControllerAnimated:YES completion:nil];
-            
-            if (weakSelf.completionBlock) {
-                NSError *error = weakSelf.connectionError;
-                weakSelf.connectionError = nil;
-                weakSelf.completionBlock(weakSelf, nil, error);
-            }
-            // Clear out the authentication challenge in memory
-            weakSelf.authenticationChallenge = nil;
-        };
-        
-        UIAlertAction *okAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"OK", @"Label: Allow the user to accept the current condition, often used on buttons to dismiss alerts")
-                                                           style:UIAlertActionStyleDefault
-                                                         handler:completion];
-        
-        [alertController addAction:okAction];
-        [self presentViewController:alertController animated:YES completion:nil];
+        [self handleConnectionError];
+        // Nil out to avoid any view did appear extra call.
+        self.connectionError = nil;
+        self.connectionErrorMessage = nil;
+    } else {
+        // We still want to report the completion block so that the necessary cleanup is done up in the call tree.
+        if (self.completionBlock) {
+            self.completionBlock(self, nil, self.connectionError);
```

Comment:
I don't think that would work in all cases, what happens if an error (that is not necessarily connection-related) occurs after appearance. The fact that we get this called before appearance is an edge case when in airplane mode or have no connection at all.